cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryProgramRunDispatcher.java

```diff
@@ -124,6 +124,7 @@ public class InMemoryProgramRunDispatcher implements ProgramRunDispatcher {
   private final LocationFactory locationFactory;
   private final RemoteClientFactory remoteClientFactory;
   private final PluginFinder pluginFinder;
+  private final ArtifactRepository noAuthArtifactRepository;
   private RemoteAuthenticator remoteAuthenticator;
   private ArtifactRepository artifactRepository;
```

Comment:
can we remove this and make it a local variable? That way there's no chance of it being 'sticky' between runs?

Reply:
fixed